tools/booster/Proxy.hs

```diff
@@ -216,74 +245,93 @@ respondEither mbStatsVar booster kore req = case req of
+                    -- We cannot call the booster internaliser without access to the server state
+                    -- Again this should not fail.
+                    let request =
+                            (emptyExecuteRequest simplified.state)
+                                { _module = mbModule
+                                , maxDepth = Just $ Depth 0
+                                }
+                    Log.logInfoNS "proxy" "Making 0-step execute request to convert back to a term/constraints form"
+                    result <- booster $ Execute request
+                    case result of
+                        Right (Execute ExecuteResult{state = finalState, logs = finalLogs}) ->
+                            pure (finalState, finalLogs)
+                        _other -> pure (s, Nothing)
+                _other -> do
+                    -- if we hit an error here, return the original
+                    Log.logWarnNS "proxy" "Unexpected failure when calling Kore simplifier, returning original term"
+                    pure (s, Nothing)
+          where
+            toSimplifyRequest :: ExecuteState -> SimplifyRequest
+            toSimplifyRequest state =
+                SimplifyRequest
+                    { state = execStateToKoreJson state
+                    , _module = mbModule
+                    , logSuccessfulSimplifications
+                    , logFailedSimplifications
+                    }
 
-        emptyExecuteRequest :: KoreJson.KoreJson -> ExecuteRequest
-        emptyExecuteRequest state =
-            ExecuteRequest
-                { state
-                , maxDepth = Nothing
-                , _module = Nothing
-                , cutPointRules = Nothing
-                , terminalRules = Nothing
-                , movingAverageStepTimeout = Nothing
-                , stepTimeout = Nothing
-                , logSuccessfulSimplifications = Nothing
-                , logFailedSimplifications = Nothing
-                , logSuccessfulRewrites = Nothing
-                , logFailedRewrites = Nothing
-                }
+            emptyExecuteRequest :: KoreJson.KoreJson -> ExecuteRequest
+            emptyExecuteRequest state =
+                ExecuteRequest
+                    { state
+                    , maxDepth = Nothing
+                    , _module = Nothing
+                    , cutPointRules = Nothing
+                    , terminalRules = Nothing
+                    , movingAverageStepTimeout = Nothing
+                    , stepTimeout = Nothing
+                    , logSuccessfulSimplifications
+                    , logFailedSimplifications
+                    , logSuccessfulRewrites
+                    , logFailedRewrites
+                    }
 
-    postExecSimplify :: Maybe Text -> API 'Res -> m (API 'Res)
-    postExecSimplify mbModule = \case
+    postExecSimplify ::
+        (Maybe Bool, Maybe Bool, Maybe Bool, Maybe Bool) -> Maybe Text -> API 'Res -> m (API 'Res)
```

Comment:
Maybe we should define `dataLogSettings = LogSettings { fieldnames... }` inside this file?

Reply:
I've decided to not do it in this PR, as it already has too much changes for my taste... should probably do it when we touch this code the next time, but I think it's not urgent.